server/lomas_server/dp_queries/dp_libraries/smartnoise_synth.py

```diff
@@ -157,69 +206,72 @@ def _get_default_constraints(
         Returns:
             table_tranformer (TableTransformer) to pre and post-process the data
         """
-        col_categories = self._get_column_by_types(
+        columns = self._get_and_check_valid_column_types(
             metadata, query_json.select_cols
         )
 
         constraints = {}
-        for col in col_categories[SSynthColumnType.PRIVATE_ID]:
-            constraints[col] = AnonymizationTransformer(SSYNTH_PRIVATE_COLUMN)
-
         nullable = query_json.nullable
-        if table_transformer_style == SSynthTableTransStyle.GAN:  # gan
-            for col in col_categories[SSynthColumnType.CATEGORICAL]:
-                constraints[col] = ChainTransformer(
-                    [LabelTransformer(nullable=nullable), OneHotEncoder()]
-                )
-            for col in col_categories[SSynthColumnType.CONTINUOUS]:
-                constraints[col] = MinMaxTransformer(
-                    lower=metadata.columns[col]["lower"],
-                    upper=metadata.columns[col]["upper"],
-                    nullable=nullable,
+        for col, col_metadata in columns.items():
+            if col_metadata.private_id:
+                constraints[col] = AnonymizationTransformer(
+                    SSYNTH_PRIVATE_COLUMN
                 )
-            for col in col_categories[SSynthColumnType.DATETIME]:
-                constraints[col] = ChainTransformer(
-                    [
-                        DateTimeTransformer(
-                            epoch=metadata.columns[col]["lower"]
-                        ),
-                        MinMaxTransformer(
-                            lower=0.0,  # because start epoch at lower bound
-                            upper=datetime_to_float(
-                                metadata.columns[col]["upper"],
-                                metadata.columns[col]["lower"],
+
+            if table_transformer_style == SSynthTableTransStyle.GAN:  # gan
+                if self._is_categorical(
+                    col_metadata
+                ):  # TODO any way of specifying cardinality? See issue #337
```

Comment:
no, no way. but it is not needed

Reply:
https://docs.smartnoise.org/synth/transforms/index.html#labeltransformer

Reply:
ok will remove the comment